Implements user search by name/email/username

dleehr opened the pull request:
This is the first half of #33 - it replaces the behavior of querying DukeDS for ALL registered users with a search form that allows searching by Name, NetID, or email address.

This change does not yet query for affiliates (just implemented in https://github.com/Duke-Translational-Bioinformatics/duke-data-service/pull/1250)

johnbradley commented:
Functionality works fine for me. 

It looks like the label and radio buttons are slightly raised:
<img width="368" alt="screen shot 2018-11-07 at 8 45 33 am" src="https://user-images.githubusercontent.com/1024463/48135283-79a3f880-e26a-11e8-8c3e-9adef80b385a.png">

The functionality between the different radio buttons is subtly different. __Name__ searches for the string anywhere in the name. While __NetID__ and __Email__ search for the exact string. I tried to search for a user by the alpha portion of their NetID and didn't find them. Perhaps we could call out this difference somehow.

Since we are allowing users to search by NetID should we include that column in the search results?
 
Assuming we could get the DukeDS team on board with adding search projects by name, it would be nice to have this search functionality in the previous step as well.

johnbradley commented on app/router.js:
```diff
@@ -23,6 +23,7 @@ Router.map(function() {
     this.route('setup-instructions');
   });
   this.route('get-token');
+  this.route('user-search');
```
Where is this route used?

dleehr replied:
That was a route I added to easily test the component. I meant to revert the commit that added it but forgot to

johnbradley commented on app/templates/components/duke-ds-user-search-fields.hbs:
```diff
@@ -0,0 +1,13 @@
+    <div class="input-group">
+      <div class="input-group-addon">Search By&nbsp;
+        {{#each kinds as |k|}}
+          <label class="input-group-text mode">{{radio-button value=k checked=kind class=k.mode}} <span>{{k.name}}</span></label>
```
What does the `mode` class do here?

dleehr replied:
It's just used to make the `label`s easy to find in tests: 

https://github.com/Duke-GCB/datadelivery-ui/blob/371e5f59e1606b71f9c2f85d79bdeb62768132b0/tests/integration/components/duke-ds-user-search-fields-test.js#L11

dleehr commented:
> Functionality works fine for me.
> 
> It looks like the label and radio buttons are slightly raised:
> <img alt="screen shot 2018-11-07 at 8 45 33 am" width="368" src="https://user-images.githubusercontent.com/1024463/48135283-79a3f880-e26a-11e8-8c3e-9adef80b385a.png">

Yeah, you're right. When comparing to the text field, the baselines are definitely not lined up.
This seems to be a side-effect of wrapping `<label>`s around radio buttons inside an input group addon. I put up a simple [JSFiddle](http://jsfiddle.net/46psurtm/) to try to tweak it, and haven't been able to find any good examples on what's missing. I think it aligns correctly if I don't use labels (Just `<span>`s or something for the text, but then they're not really clickable.

It seems like it's possible in bootstrap 4, or as a [Button with Dropdown](https://getbootstrap.com/docs/3.3/components/#input-groups-buttons-dropdowns). I'll try that and see if it works any better. Seems like it would be easier to read, write, and layout.

> The functionality between the different radio buttons is subtly different. **Name** searches for the string anywhere in the name. While **NetID** and **Email** search for the exact string. I tried to search for a user by the alpha portion of their NetID and didn't find them. Perhaps we could call out this difference somehow.

Good point, I'll add that to the label or the placeholder.

> 
> Since we are allowing users to search by NetID should we include that column in the search results?

Yes, we should. 😄 

> 
> Assuming we could get the DukeDS team on board with adding search projects by name, it would be nice to have this search functionality in the previous step as well.

Perhaps. I think that's better discussed as an issue outside this PR though. There are certainly differences between users and projects.

dleehr commented:
Reworked the markup and added username to the results. Also expanded the placeholder text a little

![search-with-username](https://user-images.githubusercontent.com/2752599/48224574-2f08a600-e368-11e8-9ab4-2a28d1bc3ec3.png)
